src/main/java/xyz/nucleoid/plasmid/game/map/template/TemplateRegion.java

```diff
@@ -20,14 +23,25 @@ public BlockBounds getBounds() {
         return this.bounds;
     }
 
+    /**
+     * Returns an optional extra data assigned to this region.
+     *
+     * @return The extra data.
+     */
+    public @Nullable CompoundTag getData() {
+        return this.data;
+    }
+
     public CompoundTag serialize(CompoundTag tag) {
         tag.putString("marker", this.marker);
         this.bounds.serialize(tag);
+        if (this.data == null) { tag.put("data", new CompoundTag()); } else { tag.put("data", this.data); }
```

Comment:
It's not super significant, but there's some inconsistency in that a data tag may be null when created, and then later become an empty compound tag when loaded from disk. It might make sense to make it stored as nullable in nbt with a `contains` check on deserialization, or it might make sense to just have the data value be nonnull and default initialized to empty tag.

Reply:
I think data should not be nullable, so new commit to fix that!